src/sql/workbench/services/objectExplorer/browser/dragAndDropController.ts

```diff
@@ -27,33 +28,56 @@ export class ServerTreeDragAndDrop implements IDragAndDrop {
 	 * Returns null, otherwise.
 	 */
 	public getDragURI(tree: ITree, element: any): string {
-		if (element instanceof ConnectionProfile) {
-			return (<ConnectionProfile>element).id;
+		if (element) {
+			if (element instanceof ConnectionProfile) {
+				return (<ConnectionProfile>element).id;
+			} else if (element instanceof ConnectionProfileGroup) {
+				return (<ConnectionProfileGroup>element).id;
+			} else if (element.nodeTypeId === 'Table' || element.nodeTypeId === 'Column') {
+				return (<TreeNode>element).id;
+			}
+			else {
+				return undefined;
+			}
 		}
-		else if (element instanceof ConnectionProfileGroup) {
-			return (<ConnectionProfileGroup>element).id;
+		else {
+			return undefined;
 		}
-		return null;
 	}
 
 	/**
 	 * Returns a label(name) to display when dragging the element.
 	 */
 	public getDragLabel(tree: ITree, elements: any[]): string {
-		if (elements[0] instanceof ConnectionProfile) {
-			return (<ConnectionProfile>elements[0]).serverName;
-		} else if (elements[0] instanceof ConnectionProfileGroup) {
-			return (<ConnectionProfileGroup>elements[0]).name;
-		} else {
+		if (elements) {
+			if (elements[0] instanceof ConnectionProfile) {
+				return (<ConnectionProfile>elements[0]).serverName;
+			} else if (elements[0] instanceof ConnectionProfileGroup) {
+				return (<ConnectionProfileGroup>elements[0]).name;
+			} else if (elements[0].label) {
+				return elements[0].label;
+			}
+			else {
+				return undefined;
+			}
+		}
+		else {
 			return undefined;
 		}
 	}
 
 	/**
 	 * Called when the drag operation starts.
 	 */
-	public onDragStart(tree: ITree, data: IDragAndDropData, originalEvent: DragMouseEvent): void {
+	public onDragStart(tree: ITree, dragAndDropData: IDragAndDropData, originalEvent: DragMouseEvent): void {
 		TreeUpdateUtils.isInDragAndDrop = true;
+		const data = dragAndDropData.getData();
+		const element = data[0];
+		if (element.nodeTypeId === 'Column' || element.nodeTypeId === 'Table') {
```

Comment:
i see you're referencing this in two parts of the code, would it make sense for this to be an enum? Or some sort of helper method